Core/source/core/pgp-key.ts

```diff
@@ -349,32 +364,53 @@ export class PgpKey {
    * Get latest self-signature date, in utc millis.
    * This is used to figure out how recently was key updated, and if one key is newer than other.
    */
-  public static lastSig = async (key: OpenPGP.key.Key): Promise<number> => {
-    await key.getExpirationTime(); // will force all sigs to be verified
-    const allSignatures: OpenPGP.packet.Signature[] = [];
+  public static lastSig = async (key: Key): Promise<number> => {
+    // "await key.getExpirationTime()" no longer works, need some alternate solution
+    // discussion is in progress: https://github.com/openpgpjs/openpgpjs/discussions/1491
+    const allSignatures: SignaturePacket[] = [];
     for (const user of key.users) {
-      allSignatures.push(...user.selfCertifications);
+      const data = { userID: user.userID, userAttribute: user.userAttribute, key: key };
+      for (const selfCert of user.selfCertifications) {
+        try {
+          // todo - add proper type in openpgp.d.ts
+          await selfCert.verify(key.keyPacket, enums.signature.certGeneric, data as any); // quick hack
```

Comment:
@IvanPizhenko I added this wile debugging imports, it's wrong as is, please have a look

Reply:
see above - need to have openpgp.js to release new version with my edits.

Reply:
https://github.com/FlowCrypt/flowcrypt-ios/issues/1377